### ISSUE TITLE
Finding position of collision between Leap Motion's hand and Game Object genereted by MapBox

### ISSUE THREAD

**Issue by asiorek** [mapbox/mapbox-unity-sdk#405]
I am new to MapBox and not really sure how to find the positions of collisions between hand from Leap Motion and Game Object generated by Map Box? For the first step I would like to print them in a Unity console.

![screen shot 2017-11-28 at 11 44 10 2](https://user-images.githubusercontent.com/8303927/33300156-313e8804-d432-11e7-9e86-240b5f10b289.png)

To generate these cylinders, I followed MeshGenerationPois example.  So, I am using VectorTileFactory and VectorLayerVisualizer, which uses ModifierStack as follows:
![screen shot 2017-11-28 at 11 53 59](https://user-images.githubusercontent.com/8303927/33300304-ea2e4bc4-d432-11e7-887d-bd626eb793a4.png)

For ColliderModifier I chose a mesh type, but I am not sure if it's a right way to do that.
I'll appreciate any suggestions.

**Comment by brnkhy** [comment]
@asiorek  Oh my god that looks incredibly cool, guess I never saw that leap otion hand in unity before :D 

-Collider modifier under game object modifier list probably won't work as the original gameobject created by that stack will be an empty game object (which contains the prefab). I don't know what kind of collider is that (mesh, sphere etc) but even if it's sphere or something, it would be infinitely small I guess. Instead, you should add collider to the prefab you're using already if they don't already have it (those aqua cylinders). 

-from what I can see, that hand is in the same space with everything else (as finger goes into the cylinder) so you should be able to get that point and/or add a collider to hand itself. With colliders on both hand and cylinders, I believe you can use unity's own collision detection (OnCollisionEnter) for it. I haven't really used it before but I'm sure you can find lots of resources on that.

**Comment by david-rhodes** [comment]
Remember that if you intend to use collision interactions, rather than simple raycasting, one of the objects needs to have a non-kinematic rigidbody, as well!

**Comment by asiorek** [comment]
I finally got it.
I added Mesh Collider and Rigidbody to each part of a hand prefab, such as: all fingers, palm and forearm.
![screen shot 2017-12-02 at 20 02 39](https://user-images.githubusercontent.com/8303927/33514686-cde440ce-d79b-11e7-932b-31092deefa39.png)

I also added Mesh Collider and Rigidbody to 3DPoiMarker prefab as well as script for detecting collisions. In Rigidbody I had to freeze the position and rotation, what prevents object from moving :)
![screen shot 2017-12-02 at 20 01 24](https://user-images.githubusercontent.com/8303927/33514682-b78bcb62-d79b-11e7-9131-e4d79ca3511a.png)

To detect this collisions I used OnTriggerEnter and print the positions. 
It works just as I wanted :)
Thank you for help!
